tasks/Task.cpp

```diff
@@ -42,8 +57,159 @@ void Task::errorHook()
+    return sensor::parse_metadata(metadata);
+}
+
+LidarScan Task::acquireData()
+{
+    size_t width = m_metadata.format.columns_per_frame;
+    size_t heigth = m_metadata.format.pixels_per_column;
+
+    // A LidarScan holds lidar data for an entire rotation of the device
+    LidarScan scan{width, heigth, m_metadata.format.udp_profile_lidar};
+
+    base::samples::IMUSensors imu_samples;
+
+    // buffer to store raw packet data
+    auto pkt_buffer = std::make_unique<uint8_t[]>(m_udp_buf_size);
+    bool complete = false;
+    while (!complete) {
+        sensor::client_state cli_state = sensor::poll_client(*m_handle);
+        // check for error status
+        if (cli_state & sensor::CLIENT_ERROR) {
+            throw std::runtime_error("Sensor client returned error state!");
+        }
+
+        // check for lidar data, read a packet and add it to the current batch
+        if (cli_state & sensor::LIDAR_DATA) {
+            if (!sensor::read_lidar_packet(*m_handle, pkt_buffer.get(), *m_packet_format)) {
+                throw std::runtime_error("Failed to read a packet of the expected size!");
+            }
+            if ((*m_scan_batcher)(pkt_buffer.get(), scan)) {
+                if (scan.complete(m_metadata.format.column_window)) {
+                    complete = true;
+                }
+            }
+        }
+        if (cli_state & sensor::IMU_DATA) {
+            sensor::read_imu_packet(*m_handle, pkt_buffer.get(), *m_packet_format);
+            float acc_x, acc_y, acc_z, av_x, av_y, av_z;
+            memcpy(&acc_x, pkt_buffer.get() + 24, sizeof(float));
+            memcpy(&acc_y, pkt_buffer.get() + 28, sizeof(float));
+            memcpy(&acc_z, pkt_buffer.get() + 32, sizeof(float));
+            memcpy(&av_x, pkt_buffer.get() + 36, sizeof(float));
+            memcpy(&av_y, pkt_buffer.get() + 40, sizeof(float));
+            memcpy(&av_z, pkt_buffer.get() + 44, sizeof(float));
+            imu_samples.acc << acc_x * 9.8, acc_y * 9.8, acc_z * 9.8;
+            imu_samples.gyro << av_x * M_PI / 180, av_y * M_PI / 180, av_z * M_PI / 180;
+            imu_samples.time = base::Time::now();
+            _imu_samples.write(imu_samples);
+        }
+    }
+    return scan;
+}
+
+void Task::convertData(LidarScan& scan)
+{
+
+    base::samples::DepthMap depth_map;
+    auto time = base::Time::now();
+    depth_map.time = time;
+    depth_map.timestamps.push_back(time);
+
+    depth_map.vertical_projection = base::samples::DepthMap::PROJECTION_TYPE::POLAR;
+    depth_map.horizontal_projection = base::samples::DepthMap::PROJECTION_TYPE::POLAR;
+
+    auto width = m_metadata.format.columns_per_frame;
+    auto height = m_metadata.format.pixels_per_column;
+
+    depth_map.horizontal_interval.push_back(0);
+    depth_map.horizontal_interval.push_back(M_PI * 2.0);
+
+    depth_map.vertical_interval.push_back(-11.25 * M_PI / (180.0));
+    depth_map.vertical_interval.push_back(11.25 * M_PI / (180.0));
```

Comment:
Arent these intervals parameters from the lidar itself? If so, I think its better to set them as properties